node/pkg/chain/helper/helper.go

```diff
@@ -184,6 +186,8 @@ func (t *ChainHelper) MakeDirectTx(ctx context.Context, contractAddressHex strin
 		}
 		return err
 	}
+	t.mu.Lock()
+	t.mu.Unlock()
```

Comment:
hmm could you pls explain why we are locking and unlocking the mutex immediately? I don't think this prevents duplicate nonce generation. How about having mutex right where we generate nonce (like you've suggested in my PR)?

Reply:
forgot to add defer keyword,
it's quite tricky since utils function is not part of the chainHelper method

Reply:
I see. But this might not work either because as far as I know wallet's transaction count gets incremented only after transaction is mined. So even if we have a mutex in transaction submission, multiple consecutive calls to this function will get the same nonce value (since we are not waiting for earlier transactions to be mined). This is the whole reason I had to implement a manual nonce manager in `vrf` and `rr` 😢

Reply:
I see, then let's stick to your plan, adding mutex into reporter.
please take this issue later
https://github.com/Bisonai/orakl/issues/1874

Reply:
quick workaround solutions are:
 - include waiting for transaction mining in the mutex -> this is how `vrf` and `rr` reporter was previously implemented and as you can guess this makes things slow
 - have a separate wallet for each submission interval -> this will work for now since we only have 2 different intervals but this is not scalable

The only scalable and fast solution I could think of for `vrf` and `rr` was to have a manual nonce manager.